engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -780,17 +780,17 @@ public void monitorHeartbeat(boolean waitForRecovery)
           //LOG.debug("{} {} {}", c.getExternalId(), currentTms - sca.createdMillis, this.vars.heartbeatTimeoutMillis);
           // container allocated but process was either not launched or is not able to phone home
           if (currentTms - sca.createdMillis > 2 * this.vars.heartbeatTimeoutMillis) {
-            LOG.error("Container {}@{} startup timeout ({} ms).", c.getExternalId(), c.host, currentTms - sca.createdMillis);
+            LOG.warn("Container {}@{} startup timeout ({} ms).", c.getExternalId(), c.host, currentTms - sca.createdMillis);
             containerStopRequests.put(c.getExternalId(), c.getExternalId());
           }
         } else {
           if (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis) {
             if (!isApplicationIdle()) {
               // Check if the heartbeat for this agent has already been missed to raise the StramEvent only once
               if (sca.lastHeartbeatMillis != -1) {
-                String info = String.format("Container %s@%s heartbeat timeout (%d%n ms).", c.getExternalId(), c.host, currentTms - sca.lastHeartbeatMillis);
-                LOG.error(info);
-                StramEvent stramEvent = new StramEvent.ContainerErrorEvent(c.getExternalId(), info, null);
+                String info = String.format("Container %s@%s heartbeat timeout  (%d%n ms).", c.getExternalId(), c.host, currentTms - sca.lastHeartbeatMillis);
+                LOG.warn(info);
+                StramEvent stramEvent = new StramEvent.ContainerErrorEvent(c.getExternalId(), info, null, StramEvent.LogLevel.WARN);
```

Comment:
The "stram" event is a user facing concept and logging is internal. For the user the container not starting or failing is probably means "error", for Apex engine it is not. Error logging should be appropriate to diagnose issues with Apex, and hence the log4j level cannot be error (it isn't an error in the engine).  If what we set on the event is called "log level", then perhaps it should be the same, but I don't see how that information is important to the user. I would suggest the event to have its own severity that should not be coupled to log4j logging.